File: oct-6th/linear_search.py
b_list = ['b', 'a', 'n', 'a', 'n', 'a', 's']

#part 1
# def linear_search(char,arr):
#   for i in range(len(arr)):
#     if char == arr[i]:
#       return i
#   return None

#part 2
def linear_search(char,arr):
  result = []
  for i in range(len(arr)):
    if char == arr[i]:
      result.append(i)
  return result


print(linear_search('a', b_list))

File: oct-5th/linearSearch.js
bList = ["b", "a", "n", "a", "n", "a", "s"]

//part 1
function linearSearch (char, arr) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === char) {
            return i
        }
    }
    return undefined
}


//part 2
function linearSearch (char, arr) {
    let resultArr = []
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === char) {
            resultArr.push(i)
        }
    }
    return resultArr
}

console.log(linearSearch("a", bList))
//[ 1, 3, 5 ]